numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -1849,7 +1849,64 @@ BOOL_fromstr(char *str, void *ip, char **endptr,
 }
 
 /**begin repeat
- * #fname = CFLOAT, CDOUBLE, CLONGDOUBLE,
+ * #fname = CFLOAT, CDOUBLE#
+ * #type = npy_cfloat, npy_cdouble#
+ */
+static int
+@fname@_fromstr(char *str, void *ip, char **endptr,
+        PyArray_Descr *NPY_UNUSED(ignore))
+{
+    double result;
+
+    result = NumPyOS_ascii_strtod(str, endptr);
+
+    @type@ output;
+
+    if (endptr && ((*endptr[0] == '+') || (*endptr[0] == '-'))) {
+        // Imaginary component specified
+
+        output.real = result;
+
+        // Reading imaginary component
+        char **prev = endptr;
+        str = *endptr;
+        result = NumPyOS_ascii_strtod(str, endptr);
+
+        if (endptr && *endptr[0] == 'j') {
+            output.imag = result;
+        }
+        else {
+            endptr = prev;
+            output.imag = 0;
+        }
+
+        // Skip j
+        ++*endptr;
```

Comment:
Shouldn't this be inside the `*endptr[0] == 'j'` branch? Maybe it does not matter, but that seems clearer to me?